pkg/controller/connectivitycheck/connectivity_check_controller.go

```diff
@@ -137,15 +142,33 @@ func (c *connectivityCheckTemplateProvider) generate(ctx context.Context, syncCo
 	}
 
 	var checks []*v1alpha1.PodNetworkConnectivityCheck
+	nodes := make(map[string]*v1.Node)
 	for _, pod := range pods {
 		if pod.Spec.NodeName == "" {
 			// network-checker pod hasn't been assigned a node yet, skip
 			continue
 		}
+		var node *v1.Node
+		var ok bool
+		if node, ok = nodes[pod.Spec.NodeName]; !ok {
+			var err error
+			node, err = c.kubeClient.CoreV1().Nodes().Get(ctx, pod.Spec.NodeName, metav1.GetOptions{})
```

Comment:
should we be using a node lister here instead of kube client? All of the other resources look to be using a lister.

Reply:
yes, changed it to use node lister now.